main.go

```diff
@@ -230,34 +196,68 @@ func (c *config) modify(f *dst.File) (*dst.File, error) {
 			return s.Fields.List[i].Names[0].Name > s.Fields.List[j].Names[0].Name
 		}
 
-		anonRevSortFunc := func(i, j int) bool {
-			return anon[i].Name > anon[j].Name
+		if c.reverse {
+			sort.Slice(s.Fields.List, revSortFunc)
+		} else {
+			sort.Slice(s.Fields.List, sortFunc)
 		}
 
+		// Sort anonymous fields
 		anonSortFunc := func(i, j int) bool {
 			return anon[i].Name < anon[j].Name
 		}
 
-		// sort anonymous fields separately
+		anonRevSortFunc := func(i, j int) bool {
+			return anon[i].Name > anon[j].Name
+		}
+
 		if c.reverse {
-			sort.Slice(s.Fields.List, revSortFunc)
-			if anon != nil {
-				sort.Slice(anon, anonRevSortFunc)
-			}
+			sort.Slice(anon, anonRevSortFunc)
 		} else {
-			sort.Slice(s.Fields.List, sortFunc)
-			if anon != nil {
-				sort.Slice(anon, anonSortFunc)
+			sort.Slice(anon, anonSortFunc)
+		}
+
+		// Append sorted anonymous fields
+		for _, f := range anon {
+			s.Fields.List = append(s.Fields.List, f.node)
+		}
+
+		// Recursively sort nested structs
+		for _, field := range s.Fields.List {
+			if structType, ok := field.Type.(*dst.StructType); ok {
+				sortStructFields(structType)
 			}
 		}
+	}
 
-		// append sorted anonymous fields (segregation)
-		if len(anon) != 0 {
-			for _, f := range anon {
-				s.Fields.List = append(s.Fields.List, f.node)
+	sortStructs := func(x dst.Node) bool {
+		t, ok := x.(*dst.TypeSpec)
+		if !ok {
+			return true
+		}
+
+		if t.Type == nil {
+			return true
+		}
+
+		if c.strct != "" && t.Name.Name == c.strct {
+			foundOne = true
+		}
+
+		s, ok := t.Type.(*dst.StructType)
+		if !ok {
+			return true
+		}
```

Comment:
The code is clearly much more readable now but any reason to get rid of the comments? I found at least some of those helpful.